test/metabase/models/setting_test.clj

```diff
@@ -142,14 +149,40 @@
     (is (= true
            (test-setting-calculated-getter)))
     (is (= true
-           (setting/user-facing-value :test-setting-calculated-getter)))))
+           (setting/user-facing-value :test-setting-calculated-getter))))
+
+  (testing "`user-facing-value` will initialize pending values"
+    (mt/discard-setting-changes [test-setting-custom-init]
+      (is (some? (setting/user-facing-value custom-init-setting))))))
 
 (deftest do-not-define-setter-function-for-setter-none-test
   (testing "Settings with `:setter` `:none` should not have a setter function defined"
     (testing "Sanity check: getter should be defined"
       (is (some? (resolve `test-setting-calculated-getter))))
     (is (not (resolve `test-setting-calculated-getter!)))))
 
+(comment
+  (#'setting/db-or-cache-value custom-init-setting)
+  )
+
+(deftest custom-init-test
+  ;; Some other pre-existing test is initializing this value, this is a stop gap until we fix its hygiene
+  (#'setting/set! custom-init-setting nil :bypass-read-only? true)
+
+  (let [get-via-db #(#'setting/db-or-cache-value :test-setting-custom-init)]
+    (testing "The value will be initialized and saved"
+      (mt/discard-setting-changes [test-setting-custom-init]
```

Comment:
It's quite tedious adding this, I wonder if we couldn't instead add an `:each` fixture here which surrounds each test with a roll-back on all the settings in the namespace. We'd need to break up some of the tests for this to work, but I think it has a certain elegance.